qiskit_experiments/library/characterization/half_angle.py

```diff
@@ -0,0 +1,123 @@
+
+
+class HalfAngle(BaseExperiment):
+    r"""A calibration experiment class to perform half angle calibration.
+
+    # section: overview
+
+        This experiment runs circuits that repeat blocks of :code:`sx - sx - y`
+        circuits inserted in a Ramsey type experiment, i.e. the full gate sequence is thus
+        :code:`Ry(π/2) - [sx - sx - y] ^ n - sx` where :code:`n` is varied, see [1]. This
+        gate sequence is designed to amplify X-Y axis errors. Such errors can occur due to
+        phase errors.
+
+        .. parsed-literal::
+
+                    ┌─────────┐┌────┐┌────┐┌───┐   ┌────┐┌────┐┌───┐┌────┐ ░ ┌─┐
+               q_0: ┤ Ry(π/2) ├┤ sx ├┤ sx ├┤ y ├...┤ sx ├┤ sx ├┤ y ├┤ sx ├─░─┤M├
+                    └─────────┘└────┘└────┘└───┘   └────┘└────┘└───┘└────┘ ░ └╥┘
+            meas: 1/════════════════════════════...═══════════════════════════╩═
+                                                                              0
+
+    # section: reference
+        .. ref_arxiv:: 1 1504.06597
+    """
+
+    __analysis_class__ = FineHalfAngleAnalysis
+
+    @classmethod
+    def _default_experiment_options(cls) -> Options:
+        r"""Default values for the fine amplitude experiment.
+
+        Experiment Options:
+            repetitions (List[int]): A list of the number of times that the gate
+                sequence :code:`[sx sx y]` is repeated.
+        """
+        options = super()._default_experiment_options()
+        options.repetitions = list(range(15))
+
+        return options
+
+    @classmethod
+    def _default_analysis_options(cls) -> Options:
+        r"""Default analysis options.
+
+        If the rotation error is very small the fit may chose a d_theta close to
+        :math:`\pm\pi`. To prevent this we impose bounds on d_theta.
+        """
+        options = super()._default_analysis_options()
+        options.normalization = True
+        options.angle_per_gate = np.pi
+        options.phase_offset = -np.pi / 2
+        options.amp = 1.0
+        options.bounds = {"d_theta": (-np.pi / 2, np.pi / 2)}
+
+        return options
+
+    def __init__(self, qubit: int):
+        """Setup a fine amplitude experiment on the given qubit.
+
+        Args:
+            qubit: The qubit on which to run the fine amplitude calibration experiment.
+        """
+        super().__init__([qubit])
+
+    @staticmethod
+    def _pre_circuit() -> QuantumCircuit:
+        """Return the preparation circuit for the experiment."""
+        return QuantumCircuit(1)
+
+    def circuits(self, backend: Optional[Backend] = None):
+        """Create the circuits for the half angle calibration experiment."""
+
+        circuits = []
+
+        for repetition in self.experiment_options.repetitions:
+            circuit = self._pre_circuit()
+
+            # First ry gate
+            circuit.rz(np.pi / 2, 0)
+            circuit.sx(0)
+            circuit.rz(-np.pi / 2, 0)
```

Comment:
I think this should be fine since it is not in the error amplified `[sx - sx - y] ^ n` sequence. Furthermore, we are checking that `sx` and `x` are parallel as Will stated.